<launch>
	<node name="detect_grasps_file" pkg="agile_grasp2" type="detect_grasps_file" output="screen">
    <!-- 0: *.pcd file, 1: sensor_msgs/PointCloud2, 2: agile_grasp2/CloudSized, 3: agile_grasp2/CloudIndexed -->
    <param name="cloud_type" value="0" />

    <!--
    <param name="cloud_file_name" value="/home/baxter/data/valkyrie/brush.pcd" />
    -->
    <param name="cloud_file_name" value="/home/andreas/data/training_march2015/train10.pcd" />
    <param name="use_importance_sampling" value="false" />

    <!-- Plotting -->
    <param name="plot_mode" value="1" /> <!-- 0: none, 1: PCL, 2: rviz -->
    <param name="only_plot_output" value="true" />

    <!-- Hand Search -->
    <!-- <rosparam param="workspace"> [0.48, 0.96, -0.38, 0.0, -0.25, 1] </rosparam> -->
    <!-- clutter -->
    <rosparam param="workspace"> [-2, 2, -2, 2, -2, 2] </rosparam>
    <rosparam param="camera_pose"> [] </rosparam>
    <param name="num_samples" value="5000" />
    <param name="num_threads" value="4" />
    <param name="nn_radius_taubin" value="0.01" />
    <param name="nn_radius_hands" value="0.1" />
    <param name="num_orientations" value="8" />

    <!-- 0: use hypotheses, 1: predict antipodal graps, 2: calculate antipodal grasps -->
    <param name="antipodal_mode" value="1" />

     <!-- 0: quadrics, 1: pcl::NormalEstimationOMP -->
    <param name="normal_estimation_method" value="1" />
    <param name="voxelize" value="true"/>
    <param name="filter_half_grasps" value="false"/>
    <param name="gripper_width_range" value="[0.03, 0.08]" />

    <!-- Hand Geometry -->
    <param name="finger_width" value="0.01" />
    <param name="hand_outer_diameter" value="0.09" />
    <param name="hand_depth" value="0.06" />
    <param name="hand_height" value="0.02" />
    <param name="init_bite" value="0.01" />

    <!-- Classifier -->
    <param name="images_directory" value="/home/andreas/data/grasp_images/" />
    <param name="model_file" value="$(find agile_grasp2)/caffe/test_1batch2.prototxt" />
    <param name="trained_file" value="$(find agile_grasp2)/caffe/bottles_boxes_cans_5xNeg.caffemodel" />
    <param name="label_file" value="$(find agile_grasp2)/caffe/labels.txt" />
    <param name="min_score_diff" value="300" />
    <param name="batch_size" value="100" />

    <!-- Clustering -->
    <param name="min_inliers" value="5" />

    <!-- Grasp Selection -->
    <param name="num_selected" value="30" />
	</node>
</launch>
